src/main/java/io/cdap/plugin/ariba/source/AribaServices.java

```
@@ -621,40 +624,74 @@ protected Request buildFetchRequest(URL endPoint, String accessToken) {
+   * @return Response
+   * @throws AribaException          AribaException
+   * @throws InterruptedException    InterruptedException
+   * @throws IOException             IOException
+   * @throws AribaRetryableException
+   */
+  public Response executeRetryableRequest(Request req)
+    throws AribaException, InterruptedException, IOException, AribaRetryableException {
+
+    LOG.debug("ARIBA URL: " + req.url());
     OkHttpClient enhancedOkHttpClient = getConfiguredClient().build();
     Response response = null;
     try {
       response = enhancedOkHttpClient.newCall(req).execute();
-    if (response.code() != HttpURLConnection.HTTP_OK && AribaUtil.isNullOrEmpty(response.message())) {
-      AribaResponseContainer responseContainer = aribaResponse(response);
-      InputStream responseStream = responseContainer.getResponseBody();
-      JsonNode jsonNode = objectMapper.readTree(responseStream);
-      String errMsg = jsonNode.get(ResourceConstants.MESSAGE).asText() != null
-        ? jsonNode.get(ResourceConstants.MESSAGE).asText() :
-        ResourceConstants.ERR_NOT_FOUND.getMsgForKey();
-      throw new AribaException(errMsg, response.code());
-    }
-    if (!isApiLimitExhausted(response)) {
+      if (response.code() != HttpURLConnection.HTTP_OK && AribaUtil.isNullOrEmpty(response.message())) {
+        AribaResponseContainer responseContainer = aribaResponse(response);
+        InputStream responseStream = responseContainer.getResponseBody();
+        JsonNode jsonNode = objectMapper.readTree(responseStream);
+        String errMsg = jsonNode.get(ResourceConstants.MESSAGE).asText() != null
+          ? jsonNode.get(ResourceConstants.MESSAGE).asText() :
+          ResourceConstants.ERR_NOT_FOUND.getMsgForKey();
+        throw new AribaException(errMsg, response.code());
+      }
       checkAndThrowException(response);
       return response;
-    } else {
-      response = enhancedOkHttpClient.newCall(req).execute();
+    } catch (IOException e) {
```

Lets remove this `catch` block and let it throw the original error to the caller.

---

Hmm, no point of try block too then ! 
Removed !